src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -271,6 +274,84 @@ public int read(byte b[], int off, int len) throws IOException {
         return readBytes(b, off, len);
     }
 
+    public byte[] readAllBytes() throws IOException {
+        long length = length();
+        if (length <= 0)
+            return super.readAllBytes();
+
+        long size = length - position();
+        if (size > (long) Integer.MAX_VALUE) {
+            String msg = String.format("Required array size too large: %d = %d - %d",
+                size, length, position());
+            throw new OutOfMemoryError(msg);
+        }
+        if (size <= 0L)
+            return new byte[0];
```

Comment:
Maybe the case where size <= 0L should also default to calling return `super.readAllBytes()`?

Reply:
Yes, that might be better.